src/DiagonalHessianApproximation.jl

```diff
@@ -0,0 +1,155 @@
+export DiagonalQN, SpectralGradient
+
+function mulSquareOpDiagonal!(res, d, v, α, β::T) where {T <: Real}
+  if β == zero(T)
+    res .= α .* d .* v
+  else
+    res .= α .* d .* v .+ β .* res
+  end
+end
+
+"""
+Implementation of the diagonal quasi-Newton approximation described in
+
+Andrei, N. 
+A diagonal quasi-Newton updating method for unconstrained optimization. 
+https://doi.org/10.1007/s11075-018-0562-7
+"""
+mutable struct DiagonalQN{T <: Real, I <: Integer, V <: AbstractVector{T}} <:
+               AbstractDiagonalQuasiNewtonOperator{T}
+  d::V # Diagonal of the operator
+  nrow::I
+  ncol::I
+  symmetric::Bool
+  hermitian::Bool
+  prod!
```

Comment:
If you want better performance, I think you should type-annotate `prod::F`, `tprod::Ft` ..., see the other operator types defined in the package.